### DISCUSSION TITLE
[7.6.0-exp1] Token balance 0

### DISCUSSION THREAD

**Discussion by Dmole** [Tribler/tribler#5775] [General]
Are we training users to not care about Token balance again?

**Comment by Dmole** [comment]
Also after getting some tokens the trust graph shows 0 historical interactions, which should not be possible.

**Comment by devos50** [comment]
A main feature of this release is that we replaced TrustChain, our accounting blockchain, with a more lightweight accounting mechanism. This new mechanism will be the backbone of our upcoming [trust ecosystem](https://github.com/Tribler/tribler/wiki/The-design-of-a-trustworthy-ecosystem-around-media-sharing) and is, unfortunately, not backwards-compatible with TrustChain records. As such, we did a reset of the token balance. While we understand that this can be frustrating for (well-behaving) users, it is important to know that this token balance _currently_ does not have large implications and is likely to only affect the stability of anonymous downloads when the network is under congestion. I will discuss this matter in our next developer meeting tomorrow since we might be able to better communicate to users why we are making these decisions.

Building a new trust ecosystem is challenging and requires us to make many iterations before we get it right. For example, TrustChain has been deployed for over two years now and we have made many small changes to its code in the process. Sometimes we come to the conclusion that a technology that we have chosen is not good enough (yet) and that requires us to perform a hard protocol/mechanism upgrade. We try to keep such decisions to an absolute minimum though.

Our plan is to replace the token balance with a ranking system that shows how well you performed in comparison with other users. While this sounds easy, there are plenty of technical challenges we have to overcome before we achieve this goal.

Also, what trust graph do you refer to in your comment? The chart that shows the evolution of your balance, or the 'trust graph' in the left menu?

**Comment by Dmole** [comment]
The left menu.

The thing is that for years the tokens have ocupied prime UI realestate without actualy being enforcable and getting reset repeatedly.

 Please burry disfunctional features in a beta section.

**Reply by ichorid** [reply]
I agree that we should hide the stuff.

**Comment by Snake883** [comment]
I started with 0 balance, nothing to download, left Tribler running, now I have a huge negative balance.
Not sure how that happened, or if its a bug.
